src/sleep.cpp

```diff
@@ -428,20 +447,34 @@ bool shouldLoraWake(uint32_t msecToWake)
 void enableLoraInterrupt()
 {
 #if SOC_PM_SUPPORT_EXT_WAKEUP && defined(LORA_DIO1) && (LORA_DIO1 != RADIOLIB_NC)
-    rtc_gpio_pulldown_en((gpio_num_t)LORA_DIO1);
+    gpio_pulldown_en((gpio_num_t)LORA_DIO1);
 #if defined(LORA_RESET) && (LORA_RESET != RADIOLIB_NC)
-    rtc_gpio_pullup_en((gpio_num_t)LORA_RESET);
+    gpio_pullup_en((gpio_num_t)LORA_RESET);
 #endif
 #if defined(LORA_CS) && (LORA_CS != RADIOLIB_NC)
-    rtc_gpio_pullup_en((gpio_num_t)LORA_CS);
+    gpio_pullup_en((gpio_num_t)LORA_CS);
 #endif
-    // Setup deep sleep with wakeup by external source
-    esp_sleep_enable_ext0_wakeup((gpio_num_t)LORA_DIO1, RISING);
+
+    if (rtc_gpio_is_valid_gpio((gpio_num_t)LORA_DIO1)) {
+        // Setup light/deep sleep with wakeup by external source
+        LOG_INFO("setup LORA_DIO1 (GPIO%02d) with wakeup by external source\n", LORA_DIO1);
+        esp_sleep_enable_ext0_wakeup((gpio_num_t)LORA_DIO1, HIGH);
+    } else {
+        LOG_INFO("setup LORA_DIO1 (GPIO%02d) with wakeup by gpio interrupt\n", LORA_DIO1);
+        gpio_wakeup_enable((gpio_num_t)LORA_DIO1, GPIO_INTR_HIGH_LEVEL);
+    }
+
 #elif defined(LORA_DIO1) && (LORA_DIO1 != RADIOLIB_NC)
-    gpio_wakeup_enable((gpio_num_t)LORA_DIO1, GPIO_INTR_HIGH_LEVEL); // SX126x/SX128x interrupt, active high
+    if (radioType != RF95_RADIO) {
+        LOG_INFO("setup LORA_DIO1 (GPIO%02d) with wakeup by gpio interrupt\n", LORA_DIO1);
+        gpio_wakeup_enable((gpio_num_t)LORA_DIO1, GPIO_INTR_HIGH_LEVEL); // SX126x/SX128x interrupt, active high
+    }
 #endif
-#ifdef RF95_IRQ
-    gpio_wakeup_enable((gpio_num_t)RF95_IRQ, GPIO_INTR_HIGH_LEVEL); // RF95 interrupt, active high
+#if defined(RF95_IRQ) && (RF95_IRQ != RADIOLIB_NC)
```

Comment:
Good call. Technically we have targets like the T-Beam that have RF95 macros for the SX1276 and also SX126X ones, so runtime discrimination probably heads off some potential issues there hopefully.

Reply:
yes, I've tested that already. Working as expected.